x-pack/legacy/plugins/apm/public/components/shared/charts/TransactionCharts/RegionMapChart/index.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  ChoroplethMap,
+  getProgressionColor,
+  getDataRange
+} from './ChoroplethMap';
+import { ColorProgressionBar } from './ColorProgressionBar';
+import { useAvgDurationByCountry } from '../../../../../hooks/useAvgDurationByCountry';
+
+const RegionMapChartToolTip: React.SFC<{
+  geojsonProperties: { [name: string]: any };
+  data?: { key: string; value: number; doc_count?: number };
+}> = ({ geojsonProperties, data }) => {
+  return (
+    <div style={{ textAlign: 'center' }}>
+      <div style={{ fontSize: 'larger' }}>{geojsonProperties.name}</div>
+      <div>
+        {i18n.translate(
+          'xpack.apm.metrics.pageLoadCharts.RegionMapChart.ToolTip.avgPageLoadDuration',
+          {
+            defaultMessage: 'Avg. page load duration:'
+          }
+        )}
+      </div>
+      <div style={{ fontWeight: 'bold', fontSize: 'larger' }}>
+        {data && data.value}ms
+      </div>
+      <div>
+        (
+        {i18n.translate(
+          'xpack.apm.metrics.pageLoadCharts.RegionMapChart.ToolTip.countPageLoads',
+          {
+            values: { docCount: data && data.doc_count },
+            defaultMessage: '{docCount} page loads'
+          }
+        )}
+        )
+      </div>
+    </div>
+  );
+};
+
+export const RegionMapChart: React.SFC = () => {
+  const { data } = useAvgDurationByCountry();
+  const [min, max] = getDataRange(data);
+
+  return (
+    <div>
+      <ChoroplethMap
+        initialMapboxOptions={{
+          zoom: 0.85,
+          center: { lng: 0, lat: 30 }
+        }}
+        initialGeojsonSource="https://vector.maps.elastic.co/files/world_countries_v1.geo.json?elastic_tile_service_tos=agree&my_app_name=ems-landing&my_app_version=7.2.0"
+        geojsonKeyProperty="iso2"
+        data={data}
+        renderTooltip={RegionMapChartToolTip}
+      />
+      <ColorProgressionBar
+        getColorStyle={getProgressionColor}
+        min={`${min}ms`}
+        max={`${max}ms`}
```

Comment:
What if the durations are very high or very low? 

We recently [added support](https://github.com/elastic/kibana/pull/41640) in our formatters for minutes and hours after [requests from support and users](https://github.com/elastic/enhancements/issues/6640). 

All the formatters expect the time in microseconds, so instead of converting to millis you should just leave it in micros. And then use our formatters to format the value (eg. `getTimeFormatter`).

Reply:
yeah, I can't see your previous code because it's squashed but I seem to remember there was a transformation of the data somewhere. Instead I think you just need to use the formatters when outputting the durations. Like in the tooltip and on the legend.

Reply:
That being said: since it'll be ms in the large majority of cases this is probably not all that important.